internal/core/plugin.go

```diff
@@ -132,6 +132,11 @@ type QuotaPlugin interface {
 	// The `serializedMetrics` return value is persisted in the Limes DB and
 	// supplied to all subsequent RenderMetrics calls.
 	Scrape(ctx context.Context, project KeystoneProject, allAZs []limes.AvailabilityZone) (result map[liquid.ResourceName]ResourceData, serializedMetrics []byte, err error)
+
+	// BuildServiceUsageRequest generates the request body payload for querying
+	// the LIQUID API endpoint /v1/projects/:uuid/report-usage
+	BuildServiceUsageRequest(project KeystoneProject, allAZs []limes.AvailabilityZone) (*liquid.ServiceUsageRequest, error)
```

Comment:
This is going to be a bit nitpicky. But since it is not a valid response to return a nil result, I'd like to see the pointer type removed:

```suggestion
	BuildServiceUsageRequest(project KeystoneProject, allAZs []limes.AvailabilityZone) (liquid.ServiceUsageRequest, error)
```

Same for BuildServiceCapacityRequest.

Historically, I'm guilty of overusing pointer types in return signatures, mostly because it's more convenient to write `return nil, err` instead of `return SomeReallyLongTypeName{}, err`. But I've been trying to cut back on this over time. Pointer types ought to be reserved for their two major uses that they express intent of: modifying a passed argument or `self`, and indicating that a value can be unpopulated (similar to `Option<>` in other languages).